iwf/workflow_state_options.py

```diff
@@ -30,11 +42,14 @@ class WorkflowStateOptions:
     execute_api_timeout_seconds: Optional[int] = None
     execute_api_retry_policy: Optional[RetryPolicy] = None
     """
-        note that the failing handling state will take the same input as the failed state
-        the type is Optional[type[WorkflowState]] but there is an issue with type hint...
-        TODO fix this type hint
+        By default, workflow would fail after execute API retry exhausted.
+        Set the state to proceed to the specified state after the execute API exhausted all retries
+        This is useful for some advanced use cases like SAGA pattern.
+        RetryPolicy is required to be set with maximumAttempts or maximumAttemptsDurationSeconds for execute API.
+        Note that the failure handling state will take the same input as the failed from state.
+        TODO the type should be the type is Optional[type[WorkflowState]] but -- there is an issue with circular import...
     """
-    execute_failure_handling_state: Optional[type] = None
+    execute_failure_handling_state_when_retry_exhausted: Optional[type] = None
```

Comment:
NOTE: python uses a different pattern as we mentioned this morning. we have wrapped a higher level model on the workflow state.